core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -487,6 +487,196 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   }
 
   /**
+   * ::Experimental::
+   * Return an RDD containing all pairs of elements with matching keys in `this` and `other`. Each
+   * pair of elements will be returned as a (k, (v1, v2)) tuple, where (k, v1) is in `this` and
+   * (k, v2) is in `other`. Performs a hash join across the cluster.
+   *
+   * It supports to join skewed data. If values of some key cannot be fit into memory, it will spill
+   * them to disk.
+   */
+  @Experimental
+  def skewedJoin[W: ClassTag](other: RDD[(K, W)]): RDD[(K, (V, W))] = {
```

Comment:
See the discuss above. I added `spark.join.skew=true` at first but removed it as per @rxin and @sryza 's suggestion.